### ISSUE TITLE
Enrich policy execution fails when enrich fields contain mapping types without index parameter

### ISSUE THREAD

**Issue by jbaiera** [elastic/elasticsearch#98019]
As of 8.9.0, a bug was introduced to the enrich policy execution code that does not allow policy execution to complete successfully if an enrich policy's `enrich_fields` contains a reference to a field in the source index that is a type that does not support [disabling indexing](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-index.html). The most common example of this kind of bug is to reference an object field, which does not support the `index: false` mapping parameter:

```
// Index with object in it
PUT jamestest-enrich-object/_doc/1
{
  "key": "1",
  "info": {
    "firstname": "James",
    "lastname": "Baiera"
  }
}

// An enrich index that references the object field in its enrich_fields property
PUT _enrich/policy/jamestest-enrich-object
{
  "match": {
    "indices": "jamestest-enrich-object",
    "match_field": "key",
    "enrich_fields": ["info"]
  }
}
```

The policy will be created, but any attempts to execute the policy will be met with the following error message:
```
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "Mapping definition for [info] has unsupported parameters:  [index : false]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "Failed to parse mapping: Mapping definition for [info] has unsupported parameters:  [index : false]",
    "caused_by": {
      "type": "mapper_parsing_exception",
      "reason": "Mapping definition for [info] has unsupported parameters:  [index : false]"
    }
  },
  "status": 400
}
```

This occurs for fields other than `object` type fields. Any fields from a source index that do not support the `index` parameter such as `binary` or `histogram` will not complete the enrich execution as described above.

This bug seems to have been introduced in 8.9.0 via https://github.com/elastic/elasticsearch/pull/96056.

Workarounds:
A work around for this issue when working with object fields is to update enrich policies to reference each desired field from an object by their full name. For instance, the above example with a firstname and lastname field would need to be referenced as such:

```
PUT _enrich/policy/jamestest-enrich-object-children
{
  "match": {
    "indices": "jamestest-enrich-object",
    "match_field": "key",
    "enrich_fields": ["info.firstname", "info.lastname"]
  }
}
```

Unfortunately, no workaround exists for enrich policies that make use of fields like `binary` or `histogram` which do not support the [`index` mapping parameter](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-index.html).

**Comment by jbaiera** [comment]
Additional issues arise when trying to execute an enrich policy that contains fields that have required parameters on them, such as the scaled float type. These parameters will need to either be captured from the source indices (possibly merged), a default will need to be provided, or will need to be removed from the resulting mappings.

**Comment by ddolcimascolo** [comment]
Hi guys,

This looks like a pretty severe regression. We have tens of enrich policies that we execute regularly and a lot of them use `enrich_fields` containing references to object fields. We do this in order to enrich target documents with the full object, without knowing the actual shape of the object. This prevents us to use the workaround of defining each inner field in the policy's `enrich_field`. Moreover, some of the referenced objects might be huge, with several level of nested objects...

I see that v8.9.1 is out but this regression isn't fixed. Is this being worked on?

Thx,
David

**Comment by dnhatn** [comment]
@ddolcimascolo We apologize for the inconvenience this issue caused. I will be working to address this tomorrow. The fix should be included in the upcoming releases, including both bug-fixes and minor updates. Thank you for your interest in Elasticsearch.

**Comment by ddolcimascolo** [comment]
Hi @dnhatn,

Thanks for your kind response. If there's anything I can do to help, just ask.

David

**Comment by dnhatn** [comment]
I've opened https://github.com/elastic/elasticsearch/pull/98683.

**Comment by ddolcimascolo** [comment]
Hi @dnhatn @jbaiera

I was reading the release notes for Elasticsearch 8.11 and found
> Update enrich execution to only set index false on fields that support it [#98038](https://github.com/elastic/elasticsearch/pull/98038) (issue: [#98019](https://github.com/elastic/elasticsearch/issues/98019))

It's not obvious if the original PR (that was since reverted) was just added again, or if there was another fix. Can you clarify?

Thx,
David

**Comment by dnhatn** [comment]
Hello @ddolcimascolo. The original PR was reverted on 8.10 only, and the fix for the original PR was included in 8.11+.

**Comment by ddolcimascolo** [comment]
Thx for the clarification
